lib/diff_file.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+class rex_ydeploy_diff_file
+{
+    private $create = [];
+    private $alter = [];
+    private $drop = [];
+
+    public function createTable(rex_sql_table $table)
+    {
+        $this->create[] = $table;
+    }
+
+    public function dropTable($tableName)
+    {
+        $this->drop[$tableName] = true;
+    }
+
+    public function ensureColumn($tableName, rex_sql_column $column)
+    {
+        $this->alter[$tableName]['ensure'][] = $column;
+    }
+
+    public function removeColumn($tableName, $columnName)
+    {
+        $this->alter[$tableName]['remove'][] = $columnName;
+    }
+
+    public function setPrimaryKey($tableName, array $primaryKey)
+    {
+        $this->alter[$tableName]['primaryKey'] = $primaryKey;
+    }
+
+    public function isEmpty()
+    {
+        return !$this->create && !$this->alter && !$this->drop;
+    }
+
+    public function getContent()
+    {
+        $content = <<<'EOL'
+<?php
+
+$sql = rex_sql::factory();
+$sql->setQuery('SET FOREIGN_KEY_CHECKS = 0');
+EOL;
```

Comment:
Wobei ich das eher unübersichtlicher finde. Auch: wenn man es dann wirklich macht und auch richtig nutzt, finde ich wichtig, dass man es direkt mit in der Datei hat. Denn wenn man dann im Up-Bereich etwas eigenes ergänzt, denkt man eher dran, es auch im Down-Bereich in umgekehrter Version zu notieren.